test/test_util.py

```diff
@@ -3,19 +3,72 @@
 from datetime import datetime
 from time import sleep
 from typing import List
-from unittest.mock import Mock
+from unittest.mock import Mock, patch
 
 import copy
 import logging
 import myhoard.util as myhoard_util
 import os
+import pymysql
 import pytest
 import random
 import subprocess
 
 pytestmark = [pytest.mark.unittest, pytest.mark.all]
 
 
+@patch("pymysql.connect")
+@pytest.mark.parametrize(
```

Comment:
we don't have the case where both ca_file and require_ssl are provided.

Reply:
I think we should have the case, to show that the behaviour works as intended, even when the input parameters are a bit overzealous. And to show that the ca_file takes precedence.

Reply:
:heavy_check_mark: